### ISSUE TITLE
Use configurable units

### ISSUE THREAD

**Issue by de-raaf-media** [jmaczan/bulma-helpers#10]
Great job. Would be nice to have configurable units. For example em or rem instead of px.

**Comment by jmaczan** [comment]
Hello Mark,

Thanks for your feedback. I think that's an interesting idea and I'm curious about your opinion - should it be possible to configure units globally (i.e. all units are ems, just like now all units are pxs) or per module? 
Feel free to contribute. :slightly_smiling_face: 

Greetings,
Jędrzej

**Comment by de-raaf-media** [comment]
Hello Jędrzej,

Well, I think the default should be rem. I think the utility first css framework tailwind has a nice implementation. Where the default is rem but there are utility classes for pixels. For instance padding:

.py-64 | padding-top: 16rem; padding-bottom: 16rem;
.py-px | padding-top: 1px; padding-bottom: 1px;
.px-0 | padding-right: 0; padding-left: 0;

Regards,
Mark

**Comment by jmaczan** [comment]
Hi Mark,

Switching the default unit is a huge breaking change and right now it's too late for it, although I can consider it in 1.0 if there will be more voices like yours :slightly_smiling_face: Currently all I can do is to add an option to build all classes for set of given units (i.e. there will be a variable to which you can assign list of units, `$units: 'vh' 'em' 'px';` and a variable with defines your main unit `$main-unit: 'px'`), without changing the default one. This kind of feature will require rebuilding the project from users, of course. :slightly_smiling_face: 
Does is sound good for you? Maybe you're willing to contribute?

Best,
Jędrzej

**Comment by de-raaf-media** [comment]
Let's first decide which direction is the best. I will then try to find time te contribute.

What about this backwards compatible solution:

Defaults to px (how it is currently done):
.has-padding-top-5

Only to rem/em/vh when specified:
.has-padding-top-5-**rem**

Regards,
Mark

**Comment by jmaczan** [comment]
Yup, sounds good. Pixels would be the most intuitive and simple unit as default unit for me, so I'd prefer to keep it as default. Your solution is exactly the same what I wanted to achieve, but it looks like I didn't express myself precisely enough. :slightly_smiling_face: So let's stick to the convention that you proposed

Best,
Jędrzej

**Comment by shime** [comment]
Hey @jmaczan!

Would you accept a patch that would let people specify their sizing and spacing configuration à la Tailwind?

This would be an example of that:

```sass
$sizing: (
  '0':  '0px',
  'px': '1px',
  '1':  '4px',
  '2':  '8px',
  '3':  '12px',
  '4':  '16px',
  '5':  '20px',
  '6':  '24px',
  '8':  '32px',
  '10': '40px',
  '12': '48px',
  '16': '64px',
  '20': '80px',
  '24': '96px',
  '32': '128px',
  '40': '160px',
  '48': '192px',
  '56': '224px',	
  '64': '256px',
)
```

This lets people use px or any other units and allows them to specify proportional values.

**Comment by jmaczan** [comment]
Hi Hrvoje,

Sounds good. Although such a change would be a breaking change and I'm not sure if everyone would like this approach. What do you think about defining variable to which you could assign one of few values (from some defined set, i.e. `default`, `array-based` for your solution, `multiple-units` for Mark's soluton), and every value would represent a different strategy of building classes - the default one could be the existing solution, but if user wants, then he may change the value of the customization variable to i.e. `array-based` and rebuild project to generate new classes and replace old ones?

**Comment by shime** [comment]
Not a fan of that approach, since having it configurable like that would add a lot of complexity to the project. 

I understand the backwards-compatibility problem, so let me know if you would want that change in the next major release (when you decide to release it). I'd be happy to help!

**Comment by jmaczan** [comment]
Great to hear that you'd like to contribute. I'll make a research and rethink your idea once again. I hope you have a good time in Hrvatska, it is such a beautiful country with friendly people. :)

Best,
Jędrzej

**Comment by shime** [comment]
Haha, I'm glad you liked it here and thank you!

**Comment by superbiche** [comment]
@jmaczan I'd also love to see this happen and would be happy to contribute to help.

**Comment by jmaczan** [comment]
@superbiche feel free to open PR :)

**Comment by jmaczan** [comment]
> I understand the backwards-compatibility problem, so let me know if you would want that change in the next major release (when you decide to release it). I'd be happy to help!

@shime The future is now! Feel free to open pull request with changes

**Comment by jmaczan** [comment]
In new version 0.4.0 I created simplified configurable units feature. https://github.com/jmaczan/bulma-helpers#customizing-unit

**Comment by de-raaf-media** [comment]
Nice!
